### PR TITLE
Align with doc on file_download control 

### PR DESCRIPTION
**Pull request by FabienLelaquais**
Addresses [doc/#575](https://github.com/Avaiga/taipy-doc/issues/575)

### CODE REVIEW

File: doc/examples/controls/file_download-dynamic-temp-file.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+# Generate the digits, save them in a CSV temporary file, then trigger a download action
+# for that file.
+def download_pi(state):
+    digits = pi(state.precision)
+    with NamedTemporaryFile("r+t", suffix=".csv", delete=False) as temp_file:
+        temp_file.write("index,digit\n")
+        for i, d in enumerate(digits):
+            temp_file.write(f"{i},{d}\n")
+    state.temp_path = temp_file.name
```
Comment (FredLL-Avaiga) [inline doc/examples/controls/file_download-dynamic-temp-file.py]: strange that temp_file is available outside the with context

File: doc/examples/controls/file_download-dynamic.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+    Adapted from https://docs.python.org/3/library/decimal.html
+    """
+    saved_precision = getcontext().prec  # Save precision
+    getcontext().prec = precision
+    three = Decimal(3)  # substitute "three=3.0" for regular floats
+    lasts, t, s, n, na, d, da = 0, three, 3, 1, 0, 0, 24
+    while s != lasts:
+        lasts = s
+        n, na = n + na, na + 8
```
Comment (FredLL-Avaiga) [inline doc/examples/controls/file_download-dynamic.py:35]: this is not really easy to read ?

File: src/taipy/gui/gui_actions.py [hunk 3]
```diff
@@ -24,9 +24,33 @@ def download(
...
         name: File name for the content on the client browser (defaults to content name).
-        on_action: Callback function to call when the download ends.
+        on_action: Callback function (or callback name) to call when the download ends. See below.
+
+    ## Notes:
+
+    - *content*: this parameter can hold several values depending on your use case:
+        - a string: the value must be an existing path name to the file that gets downloaded or
+          the URL to the resource you want to download.
+        - a readable buffer (such as an `io.StringIO`): if the size of the buffer is smaller than
```
Comment (FredLL-Avaiga) [inline src/taipy/gui/gui_actions.py]: io.StringIO is not valid but bytes(io.StringIO().getValue()) is ?

File: src/taipy/gui/gui_actions.py [hunk 4]
```diff
@@ -24,9 +24,33 @@ def download(
...
+           used to determine the [MIME type](https://en.wikipedia.org/wiki/Media_type) of the
+           buffer content, and the download is performed using a generated "data:" URL with
+           the relevant type, and a base64-encoded version of the buffer content.<br/>
+           If the buffer is too large, its content is transferred after saving it in a temporary
+           server file.
+      - *on_action*: TODO CHECK<br/>
+           this callback is triggered when the transfer of the content is achieved.</br>
+           In this function, you can perform any clean up operation that could be required after
+           the download is performed.<br/>
+           This callback expects [] paramters:
```
Comment (FredLL-Avaiga) [inline src/taipy/gui/gui_actions.py:48]: [] ?

File: src/taipy/gui/viselements.json [hunk 5]
```diff
@@ -1127,7 +1127,7 @@
           {
             "name": "page",
             "type": "dynamic(str)",
-            "doc": "The page to show as the content of the block (page name if defined or an URL in an <i>iframe</i>).<br/>This should not be defined if <i>partial</i> is set."
+            "doc": "The page to show as the content of the block (page name if defined or a URL in an <i>iframe</i>).<br/>This should not be defined if <i>partial</i> is set."
```
Comment (FredLL-Avaiga) [inline src/taipy/gui/viselements.json:1130]: an URL sounds better ...

**Review by FredLL-Avaiga — APPROVED** [review]

File: doc/examples/controls/file_download-dynamic-temp-file.py (hunk 1 above)
Comment (FabienLelaquais) [reply doc/examples/controls/file_download-dynamic-temp-file.py]: Yeah... I thought it was clearer, but I'll change that.

File: doc/examples/controls/file_download-dynamic.py (hunk 2 above)
Comment (FabienLelaquais) [reply doc/examples/controls/file_download-dynamic.py:35]: That comes straight from the Python doc. I won't change that.

File: src/taipy/gui/gui_actions.py (hunk 4 above)
Comment (FabienLelaquais) [reply src/taipy/gui/gui_actions.py:48]: That's right after a TODO block...

File: src/taipy/gui/viselements.json (hunk 5 above)
Comment (FabienLelaquais) [reply src/taipy/gui/viselements.json:1130]: See https://www.techtarget.com/whatis/feature/Which-is-correct-a-URL-or-an-URL

File: src/taipy/gui/gui_actions.py (hunk 3 above)
Comment (FabienLelaquais) [reply src/taipy/gui/gui_actions.py]: Let me make this more clear.
Thx

**Pull request merged**